src/main.rs

```diff
@@ -56,6 +57,37 @@ impl Color {
     }
 }
 
+pub enum BackgroundMode {
+    /// Scale image to make the shortest dimension (i.e. height or width)
+    /// fit it's container pertaining aspect ratio.
+    Fill,
+
+    /// Scale image width to fit container width pertaining aspect ratio.
+    Fit,
+
+    /// Scale height and width to fit container's. May create distortion.
+    Stretch,
+
+    /// Do not scale image and place to center.
+    Center,
+
+    /// Do not scale image and create repeated image forming tile-pattern.
+    Tile,
+}
+
+impl BackgroundMode {
+    fn from_str(s: &str) -> BackgroundMode {
```

Comment:
Consider implementing [`FromStr`](https://doc.rust-lang.org/1.11.0/std/str/trait.FromStr.html) here, as it is a standard trait for this kind of task.